src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -2532,6 +2532,29 @@ static QVariant fcnConvexHull( const QVariantList &values, const QgsExpressionCo
   QVariant result = !geom.isNull() ? QVariant::fromValue( geom ) : QVariant();
   return result;
 }
+
+static QVariant fcnMinimalCircle( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
+{
+  QgsGeometry fGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  QgsPointXY center;
+  double radius;
+  unsigned int segments = 36;
+  if ( values.length() == 2 )
+    segments = QgsExpressionUtils::getDoubleValue( values.at( 1 ), parent );
+  QgsGeometry geom = fGeom.minimalEnclosingCircle( center, radius, segments );
+  QVariant result = !geom.isNull() ? QVariant::fromValue( geom ) : QVariant();
+  return result;
+}
+
+static QVariant fcnOrientedBBox( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
+{
+  QgsGeometry fGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  double area, angle, width, height;
+  QgsGeometry geom = fGeom.orientedMinimumBoundingBox( area, angle, width, height );
```

Comment:
I'm wondering, should this method have default parameters so one doesnt need to create dummy variables in cases where they are not interesting?

Reply:
Theses parameters are an outputs.

Reply:
I know, that doesn't change anything, does it?

Reply:
No, you're right.

I will push an overloaded methods for both.

Reply:
I think you can just add default parameters, no reason for an overloaded method?

Reply:
Ah wait, you are right, that only works with const references.

Reply:
Yes. I have tried it without success. 
Maybe pointers, but...
I have see this: [std::optional](http://en.cppreference.com/w/cpp/experimental/optional) (not tested).

I'm trying that:
qgsgeometry.cpp:
`
QgsGeometry QgsGeometry::minimalEnclosingCircle(unsigned int segments) const
{
    QgsPointXY center;
    double radius;
    return minimalEnclosingCircle( center, radius, segments );

}`

qgsexpressionfunction.cpp:
`
static QVariant fcnMinimalCircle( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
{
  QgsGeometry fGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
  unsigned int segments = 36;
  if ( values.length() == 2 )
    segments = QgsExpressionUtils::getDoubleValue( values.at( 1 ), parent );
  QgsGeometry geom = fGeom.minimalEnclosingCircle( segments );
  QVariant result = !geom.isNull() ? QVariant::fromValue( geom ) : QVariant();
  return result;
}`

(same for orientedBBox)

Reply:
I tested it and it works. @m-kuhn are you okay with that? Can I push it?

Reply:
An overloaded function sounds perfectly fine here.

Or alternatively you could return a struct

```cpp
struct MinimalCircleResult
{
  QgsGeometry geometry;
  double radius;
  double xCenter;
  double yCenter;
};
```

Personally I'm not very much a big fan of out parameters ;)

Reply:
As you want. I rewrite it with struct ;)

Reply:
Be warned that this was a very personal statement :P

Reply:
I'm sorry, I can't find the solution to use a structure. I'm dealing with incomplete types...